policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -319,7 +395,8 @@ impl Namespace {
         for service in self.service_port_routes.values_mut() {
             service.delete(gknn);
         }
-        for routes in self.service_routes.values_mut() {
+
+        for routes in self.service_routes.values_mut().flatten() {
```

Comment:
I think there needs to be something here to remove routes from service_routes if it becomes empty.  this is so that if all HttpRoutes for a service are deleted that GrpcRoutes can then be attached.

Reply:
Updated. If you would re-review whenever you have a chance?